lstchain/reco/dl1_to_dl2.py

```diff
@@ -417,3 +429,85 @@ def apply_models(dl1, classifier, reg_energy, reg_disp_vector, custom_config={})
     dl2['gammaness'] = probs
     return dl2
 
+
+
+def get_source_dependent_parameters(data, config={}):
+
+    """Get parameters for source-dependent analysis .
+
+    Parameters:
+    -----------
+    data: Pandas DataFrame
+    config: dictionnary containing configuration
+    
+    """
+
+    src_dep_params = pd.DataFrame(index=data.index)
+
+    is_simu = 'mc_type' in data.columns
+    
+    if is_simu:
+        if (data['mc_type'] == 0).all():
+            data_type = 'mc_gamma'
+        else:
+            data_type = 'mc_proton'
+    else:
+        data_type = 'real_data'
+    
+    expected_src_pos_x_m, expected_src_pos_y_m = get_expected_source_pos(data, data_type, config)
+
+    src_dep_params['expected_src_x'] = expected_src_pos_x_m
+    src_dep_params['expected_src_y'] = expected_src_pos_y_m
+    
+    src_dep_params['dist'] = np.sqrt((data['x'] - expected_src_pos_x_m)**2 + (data['y'] - expected_src_pos_y_m)**2)
+    src_dep_params['time_gradient_from_source'] = data['time_gradient'] * np.sign(data['x'] - expected_src_pos_x_m)
+    src_dep_params['skewness_from_source'] = data['skewness'] * np.sign(data['x'] - expected_src_pos_x_m)
+    
+    disp, miss = camera_to_shower_coordinates(expected_src_pos_x_m, expected_src_pos_y_m, data['x'], data['y'], data['psi'])
+    alpha = np.rad2deg(np.arctan2(miss, disp))
```

Comment:
I think you can get around these different cases here by doing `np.rad2deg(np.arctan(np.abs(miss / disp))`

Reply:
So deliberately throwing away any sign information should give you the angle in [0, 90°]

Reply:
thanks, done!